dali/pipeline/workspace/workspace.h

```diff
@@ -145,12 +145,54 @@ class WorkspaceBase : public ArgumentWorkspace {
   /**
    * @brief Returns the number of inputs.
    */
-  inline int NumInput() const { return input_index_map_.size(); }
+  inline int NumInput() const {
+    return input_index_map_.size();
+  }
 
   /**
    * @brief Returns the number of outputs.
    */
-  inline int NumOutput() const { return output_index_map_.size(); }
+  inline int NumOutput() const {
+    return output_index_map_.size();
+  }
+
+  /**
+   * Returns shape of input at given index
+   * @return TensorShape<> for SampleWorkspace, TensorListShape<> for other Workspaces
+   */
+  template <typename Backend>
+  auto GetInputShape(int input_idx) const {
+    const auto& input = *InputHandle(input_idx, Backend{});
+    return input.shape();
+  }
+
+  /**
+   * Returns batch size for given input
+   */
+  int GetInputBatchSize(int input_idx) const {
```

Comment:
Btw, I just found there is `NumInputAtIdx` and `NumOutputAtIdx` in the host workspace, which probably does something similar. Can you check if they are needed and if not remove them/replace with the API that we want to support?

Reply:
Done